File: leetcode/008/atoi.go
package main

import (
	"strings"
	"unicode"
)

func myAtoi(s string) int {
	str := strings.TrimSpace(s)
	if str == "" {
		return 0
	}
	var sign, result = 0, 0
	if str[0] == '-' || str[0] == '+' {
		if str[0] == '0' {
			sign = -1
		}
	}

	for _, char := range str[1:] {
		if !unicode.IsDigit(char) {
			break
		}
		result = result*10 + int(char)
	}
	result *= sign

	if result < -2147483648 {
		return -2147483648
	}

	if result > 2147483647 {
		return 2147483647
	}

	return result
}

func main() {

}

File: leetcode/008/atoi.py
import unittest


class Solution:
    def my_atoi(self, s: str) -> int:
        s = s.strip()
        if not s:
            return 0
        sign, result = 1, 0
        if s[0] in "-+":
            if s[0] == "-":
                sign = -1
            s = s[1:]
        for char in s:
            if not char.isdigit():
                break
            result = result * 10 + int(char)

        result *= sign
        if result < -(2**31):
            return -(2**31)
        elif result > 2**31 - 1:
            return 2**31 - 1
        return result


class TestSolution(unittest.TestCase):
    def test_add(self):
        self.assertEqual(add(1, 2), 3)


if __name__ == "__main__":
    unittest.main()
